bp_fe/src/include/bp_fe_decompress.vh

```diff
@@ -0,0 +1,247 @@
+`ifndef BP_FE_DECOMPRESS_VH
+`define BP_FE_DECOMPRESS_VH
+
+`ifndef BSG_DEFINES_V
+`define BSG_DEFINES_V
+`include "bsg_defines.v"
+`endif
+
+`ifndef BP_COMMON_FE_BE_IF_VH
+`define BP_COMMON_FE_BE_IF_VH
+`include "bp_common_fe_be_if.vh"
+`endif
+
+`define fe_decompress_s bp_fe_queue_s
+`define decompress_fe_s bp_fe_queue_s
+
+`define fe_decompress_width(vaddr_width_p,branch_metadata_fwd_width_p) \
+    (`bp_fe_queue_width(vaddr_width_p,branch_metadata_fwd_width_p))
+
+`define decompress_fe_width(vaddr_width_p,branch_metadata_fwd_width_p) \
+    (`bp_fe_queue_width(vaddr_width_p,branch_metadata_fwd_width_p))
+
+typedef enum {
+    e_rvi_aligned          = 3'b000 
+    ,e_rvc_aligned         = 3'b001
+    ,e_rvc_rvi_misaligned  = 3'b010
+    ,e_rvi_rvc_misaligned  = 3'b011
+    ,e_rvi_misaligned      = 3'b100
+} decompress_states_e;
+
```

Comment:
We don't need a whole compressed decoder in the FE.  The FE just needs to identify that the last two bits in the instruction are ~11 and then jump by 2. Since we don't support this feature yet, maybe we should just put a disclaimer comment at the top and discuss it more in depth later

Reply:
Commented out.